gc.c

```diff
@@ -689,29 +689,27 @@ typedef struct RVALUE {
             VALUE v3;
         } values;
     } as;
+} RVALUE;
 
-    /* Start of RVALUE_OVERHEAD.
-     * Do not directly read these members from the RVALUE as they're located
-     * at the end of the slot (which may differ in size depending on the size
-     * pool). */
-#if RACTOR_CHECK_MODE
+/* These members ae located at the end of the slot that the object is in. */
+#if RACTOR_CHECK_MODE || GC_DEBUG
+struct rvalue_overhead {
+# if RACTOR_CHECK_MODE
     uint32_t _ractor_belonging_id;
-#endif
-#if GC_DEBUG
+# endif
+# if GC_DEBUG
     const char *file;
     int line;
-#endif
-} RVALUE;
+# endif
+} __attribute__((aligned(sizeof(VALUE))));
```

Comment:
May not be portable.

Reply:
I changed it to use a union in a struct instead. That should be portable.